From 127b7874afd3dfd3ee2049514e8a7ae616daaab9 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolabs.io>
Date: Fri, 13 Jan 2017 10:17:40 +0100
Subject: [PATCH 33/55] direct3d11: create a shader resource view for each
 slice of the decoder texture

This way it can be passed directly to the pixel shader, straight out of the
decoder.
---
 modules/video_output/win32/direct3d11.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/modules/video_output/win32/direct3d11.c b/modules/video_output/win32/direct3d11.c
index bc69f7b879..a33863dcf7 100644
--- a/modules/video_output/win32/direct3d11.c
+++ b/modules/video_output/win32/direct3d11.c
@@ -679,6 +679,9 @@ static picture_pool_t *Pool(vout_display_t *vd, unsigned pool_size)
     unsigned          picture_count = 0;
     HRESULT           hr;
 
+    const d3d_format_t *cfg = GetOutputFormat(vd, vd->fmt.i_chroma, 0, true, false);
+    assert(cfg == vd->sys->picQuadConfig);
+
     ID3D10Multithread *pMultithread;
     hr = ID3D11Device_QueryInterface( vd->sys->d3ddevice, &IID_ID3D10Multithread, (void **)&pMultithread);
     if (SUCCEEDED(hr)) {
@@ -721,6 +724,9 @@ static picture_pool_t *Pool(vout_display_t *vd, unsigned pool_size)
         picsys->slice_index = picture_count;
         picsys->context = vd->sys->d3dcontext;
 
+        if (AllocateShaderView(vd, vd->sys->picQuadConfig, texture, picture_count, picsys->resourceView) != VLC_SUCCESS)
+            goto error;
+
         picture_resource_t resource = {
             .p_sys = picsys,
             .pf_destroy = DestroyDisplayPoolPicture,
-- 
2.11.0

